### ISSUE TITLE
design issues trying to update a system where the subscription has lapsed

### ISSUE THREAD

**Issue by andreasn** [cockpit-project/cockpit#9926]
I ran 20 usability tests together with Sara Chizari a couple of weeks back. One of the tasks was to update a system that wasn't currently subscribed, and a majority of the participants had a lot of problems finishing that task.

The participants kept hunting for the update button without success, and only some realized that the system needed to be subscribed first.

I suggest keeping the banner, but also show the Update buttons in a disabled state, with hints on what needs to be done upon hover.

**Comment by andreasn** [comment]
![mockup](https://raw.githubusercontent.com/cockpit-project/cockpit-design/4ef1114da8257b6c1fee24f4b962a7f33867cfb1/software-updates/software-updates-packages-registration_lapse.png)

**Comment by martinpitt** [comment]
I'm fine with either proposal. The empty state pattern is more rigid and maybe less confusing. In most cases you should not actually be able to query any actual updates when the system is not subscribed - this only "worked" in the usability test because you were using some mock updates. But having additional repositories which don't require subscriptions might not be that rare - but if we allowed the user to only install those, we might "train" them that it's seemingly okay to ignore the "not subscribed" error.

So in summary, I currently lean towards the empty state pattern.

**Comment by garrett** [comment]
Yeah, pure empty state, for the reasons @martinpitt mentioned (lack of a way to properly query package updates) :+1: 

Note that we should have a warning on the overview page (as well as a way to subscribe) too, if it doesn't exist already.

**Comment by andreasn** [comment]
> Note that we should have a warning on the overview page (as well as a way to subscribe) too, if it doesn't exist already.

Yup, we have that already.

![screenshot from 2018-09-10 13-11-22](https://user-images.githubusercontent.com/209895/45294257-09dbf080-b4fb-11e8-9cdd-f78d7e0aad5f.png)

**Comment by garrett** [comment]
@andreasn: Yeah, I wasn't sure, but thought we might. Great! Thanks for the confirmation!

**Comment by martinpitt** [comment]
Reproducer:
 * Start our rhel-7-6 image; cockpit's "Software Updates" will show the empty state pattern
 * Enable third-party repo that is not covered by subscriptions:

       cd /etc/yum.repos.d/; curl -O https://copr.fedorainfracloud.org/coprs/g/cockpit/cockpit-preview/repo/epel-7/group_cockpit-cockpit-preview-epel-7.repo
 * Now updates page switches to combined banner/update list mode.

So I'll take away that combined mode.
